src/usr/share/gpgmailer/gpgkeyverifier.py

```diff
@@ -198,17 +186,18 @@ def _calculate_recipient_info(self, loop_current_time):
                 elif expiration_data['expiring_soon']:
                     expiring_soon_messages.append(expiration_data['warning_message'])
 
-                    valid_key_fingerprints.append(self.email_dicts[recipient_email]['fingerprint'])
-                    valid_recipient_emails.append(recipient_email)
+                    valid_key_fingerprints.append(fingerprint)
+                    for email in self.recipient_keys[fingerprint]['email_list']:
+                        valid_recipient_emails.append(email)
```

Comment:
I'm not sure this has to be added in a loop. (Other places probably have this problem too.)

Reply:
Switched this back to list.extend in all cases.